specification/app/resource-manager/Microsoft.App/preview/2023-04-01-preview/Jobs.json

```diff
@@ -632,6 +624,24 @@
       }
     }
   },
+  "parameters": {
+    "JobNameParameter": {
+      "name": "jobName",
+      "in": "path",
+      "description": "Job Name",
+      "required": true,
+      "type": "string",
+      "pattern": "^[-\\w\\._\\(\\)]+$"
+    },
+    "JobExecutionNameParameter": {
+      "name": "jobExecutionName",
+      "in": "path",
+      "description": "Job execution name.",
+      "required": true,
+      "type": "string",
+      "pattern": "^[-\\w\\._\\(\\)]+$"
+    }
+  },
```

Comment:
please add `"x-ms-parameter-location": "method"` if they are not parameters of client.

Reply:
Shouldn't this will be a breaking change? The API  exists in the previous version already. 

https://github.com/Azure/azure-rest-api-specs/blob/277cfe51e02796b60026dd1214ac6617f463472e/specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json#L587

And we also have a bother object, ContainerApp has the same method. If we change this it will make it inconsistence too.

https://github.com/Azure/azure-rest-api-specs/blob/277cfe51e02796b60026dd1214ac6617f463472e/specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ContainerApps.json#L391

And at the end, I don't understand why this will have a better experience, when the change you suggest, the SDK will be

jobCollection.ListSecret(jobName)

but now, I think it is 

job.ListSerect()_

The current one seems to have better experience.

Reply:
> Shouldn't this will be a breaking change? The API exists in the previous version already.
> 
> https://github.com/Azure/azure-rest-api-specs/blob/277cfe51e02796b60026dd1214ac6617f463472e/specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/Jobs.json#L587
> 
> And we also have a bother object, ContainerApp has the same method. If we change this it will make it inconsistence too.
> 
> https://github.com/Azure/azure-rest-api-specs/blob/277cfe51e02796b60026dd1214ac6617f463472e/specification/app/resource-manager/Microsoft.App/preview/2022-11-01-preview/ContainerApps.json#L391
> 
> And at the end, I don't understand why this will have a better experience, when the change you suggest, the SDK will be
> 
> jobCollection.ListSecret(jobName)
> 
> but now, I think it is
> 
> job.ListSerect()_
> 
> The current one seems to have better experience.

When you move the parameter from operation local to global, there will be a implicit change that the parameter become global parameter, then it will be treated as a client parameter for SDK which is a breaking change for SDK. You may see the breakings [here](https://github.com/azure-sdk/azure-sdk-for-go/pull/5810/files#diff-5c2d2be74f03e24cd540936fa57f9312c7c279017fe6324c1353f04bd42d71eaR6-R17).